### PR TITLE
Fix memory leak, refactor spectrumdsp, small amounts of MVC cleanup

### PR DESCRIPTION
**Pull request by JustAnEE**
Memory Leak in SpectrumDSP Class fixed by refactor. 
Added virtual destructors to base objects in MVC
Other misc stuff

### CODE REVIEW

File: Spectrum-Analyzer/Spectrum-Analyzer/Src/Dsp/SpectrumTypes/FFTCL.hpp [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+   // Public constructors / destructors 
+   public:
+      FFTCL(int iBufferSize_, int iSampleRate_);
+      ~FFTCL(); 
+
+   // Public methods 
+   public:
+      void SetSampleData(float* pafSampleData_);
+      void FFT(); 
+      //void IFFT();
```
Comment (DanSop) [inline Spectrum-Analyzer/Spectrum-Analyzer/Src/Dsp/SpectrumTypes/FFTCL.hpp:17]: is this necessary?

File: Spectrum-Analyzer/Spectrum-Analyzer/Src/Dsp/SpectrumTypes/MagnitudeSpectrum.cpp [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+{
+
+   return; 
+}
+
+
+void MagnitudeSpectrum::CalculateSpectrum()
+{
+   
+   // We can just interleave the data,
```
Comment (DanSop) [inline Spectrum-Analyzer/Spectrum-Analyzer/Src/Dsp/SpectrumTypes/MagnitudeSpectrum.cpp:14]: um ecuse me muiltipleline coments shoud be likke
```c++
/*
 * i have aids
 * please help me
 */
```

File: Spectrum-Analyzer/Spectrum-Analyzer/Src/MVC/Cpp/SpectrumController.cpp [hunk 3]
```diff
@@ -81,7 +78,7 @@ void SpectrumController::handleMouseClick(GLFWwindow* window, int button, int ac
                 IModel->setCurrentListMenuOption(clickedPlot->getMethodFlag());
             }
             else {
-                cout << "ERROR IN SpectrumController::handleMouseClick.     IModel->getCurrentListMenuID() not a valid ID.";
+                std::cout << "ERROR IN SpectrumController::handleMouseClick.     IModel->getCurrentListMenuID() not a valid ID.";
```
Comment (DanSop) [inline Spectrum-Analyzer/Spectrum-Analyzer/Src/MVC/Cpp/SpectrumController.cpp:81]: this string is long - can we define these errors somewhere?

File: Spectrum-Analyzer/Spectrum-Analyzer/Src/MVC/Hpp/SpectrumController.hpp [hunk 4]
```diff
@@ -9,15 +9,22 @@
...
 
-using namespace std;
+
+typedef enum ControllerStateEnum
+{
+	START   = 0,
+	READING = 1,
+	PROCESSING = 2,
+	HALT = 3 
+}ControllerStateEnum;
```
Comment (DanSop) [inline Spectrum-Analyzer/Spectrum-Analyzer/Src/MVC/Hpp/SpectrumController.hpp:19]: } ControllerStateEnum;

File: Spectrum-Analyzer/Spectrum-Analyzer/Src/Dsp/SpectrumTypes/MagnitudeSpectrum.cpp (hunk 2 above)
Comment (Nolan-Rewega) [reply Spectrum-Analyzer/Spectrum-Analyzer/Src/Dsp/SpectrumTypes/MagnitudeSpectrum.cpp:14]: Please refrain from using such language on the public GitHub repo.

File: Spectrum-Analyzer/Spectrum-Analyzer/Src/MVC/Cpp/SpectrumController.cpp (hunk 3 above)
Comment (Nolan-Rewega) [reply Spectrum-Analyzer/Spectrum-Analyzer/Src/MVC/Cpp/SpectrumController.cpp:81]: No

**Review by DanSop — COMMENTED** [review]
Solid stuff - this cleans it up a lot

File: Spectrum-Analyzer/Spectrum-Analyzer/Src/MVC/Hpp/SpectrumController.hpp (hunk 4 above)
Comment (Nolan-Rewega) [reply Spectrum-Analyzer/Spectrum-Analyzer/Src/MVC/Hpp/SpectrumController.hpp:19]: Nice catch!

**Review by Nolan-Rewega — APPROVED** [review]
Nice work!

File: Spectrum-Analyzer/Spectrum-Analyzer/Src/MVC/Cpp/SpectrumController.cpp [hunk 5]
```diff
@@ -137,7 +134,7 @@ void SpectrumController::handleListMenu(std::string ListID, int optionNum) {
     else if (ListID == "Window") { IModel->getSelectedPlot()->setWindowFlag(optionNum); }
     else if (ListID == "Method") { IModel->getSelectedPlot()->setMethodFlag(optionNum); }
     else {
```
Comment (DanSop) [inline Spectrum-Analyzer/Spectrum-Analyzer/Src/MVC/Cpp/SpectrumController.cpp:136]: We should change, does not look great

Comment (JustAnEE) [reply Spectrum-Analyzer/Spectrum-Analyzer/Src/MVC/Cpp/SpectrumController.cpp:136]: needs some work....

**Comment by DanSop** [comment]
Nice work overall!

File: Spectrum-Analyzer/Spectrum-Analyzer/Src/Dsp/Cpp/spectrumdsp.cpp [hunk 6]
```diff
@@ -207,59 +193,6 @@ void SpectrumDSP::ifft(fftwf_complex* in, fftwf_complex* out)
...
-    fft(shifted_input, y);
-
-    return y;
-}
-
-fftwf_complex* SpectrumDSP::fft_shift() 
-{
-    fftwf_complex* x = (fftwf_complex*)fftwf_malloc(sizeof(fftwf_complex) * (iMyBufferSize));
-
-        for (int k = 0; k < iMyBufferSize; k++) {
```
Comment (DanSop) [inline Spectrum-Analyzer/Spectrum-Analyzer/Src/Dsp/Cpp/spectrumdsp.cpp:223]: Open the bracket

**Pull request merged**